plugins/chain_plugin/trx_finality_status_processing.cpp

```diff
@@ -1,12 +1,265 @@
+                        ("storage", storage_to_free)
+                        ("remove_entries", remove_trxs.size()));
+            break;
+         }
+         else {
+            const auto block_num = cbh::num_from_id(oldest_block_iter->block_id);
+            if (earliest_block == finality_status::no_block_num) {
+               earliest_block = block_num;
+            }
+            latest_block = block_num;
+            const auto block_timestamp = oldest_block_iter->block_timestamp;
+            for (; oldest_block_iter != block_indx.end() && cbh::num_from_id(oldest_block_iter->block_id) == block_num; ++oldest_block_iter) {
+               reduce_storage(*oldest_block_iter);
+            }
+            for (auto oldest_failure_iter = status_expiry_indx.upper_bound( std::make_tuple( false, block_timestamp.to_time_point() ) );
+                 oldest_failure_iter != oldest_failure_end;
+                 ++oldest_failure_iter) {
+               reduce_storage(*oldest_failure_iter);
+            }
+         }
+      }
+
+      for (const auto& trx_id : remove_trxs) {
+         _storage.erase(trx_id);
+      }
+
+      if (earliest_block != finality_status::no_block_num) {
+         ilog( "Finality Status dropped ${trx_count} transactions, which were removed from block # ${block_num_start} to block # ${block_num_end}",
+               ("trx_count", remove_trxs.size())("block_num_start", earliest_block)("block_num_end", latest_block) );
+      }
+      else {
+         ilog( "Finality Status dropped ${trx_count} transactions, all were failed transactions", ("trx_count", remove_trxs.size()) );
+      }
+   }
+
+   trx_finality_status_processing::chain_state trx_finality_status_processing::get_chain_state() const {
+      return { .head_id = _my->_head_block_id, .irr_id = _my->_irr_block_id, .last_tracked_block_id = _my->_last_tracked_block_id };
+   }
+
+   std::optional<trx_finality_status_processing::trx_state> trx_finality_status_processing::get_trx_state( const chain::transaction_id_type& id ) const {
+      trx_finality_status_processing::trx_state state;
+      auto iter = _my->_storage.find(id);
+      if (iter == _my->_storage.index().cend()) {
+         return {};
+      }
+
+      state.block_id == iter->block_id;
+      state.block_timestamp = iter->block_timestamp;
+      if (iter->forked_out) {
+         state.status = "FORKED OUT";
+      }
+      else if (iter->block_id == chain::block_id_type{}) {
+         if (fc::time_point::now() >= iter->trx_expiry) {
+            state.status = "FAILED";
+         }
+         else {
+            state.status = "LOCALLY APPLIED";
+         }
+      }
+      else {
+         const auto block_num = cbh::num_from_id(iter->block_id);
+         const auto lib = cbh::num_from_id(_my->_irr_block_id);
+         if (block_num > lib) {
+            state.status = "IN BLOCK";
+         }
+         else {
+            state.status = "IRREVERSIBLE";
+         }
```

Comment:
I wonder if it might make it slightly easier for external tools to parse these if they were all one word.
`FORKED_OUT`, `FAILED`, `LOCALLY_APPLIED`, `IN_BLOCK`, `IRREVERSIBLE`